router/core/context.go

```diff
@@ -664,6 +657,7 @@ func NewSubgraphResolver(subgraphs []Subgraph) *SubgraphResolver {
 			Url:       subgraphs[i].Url,
 			UrlString: subgraphs[i].UrlString,
 		}
+		// TODO: What happens here if the URL is duplicated? CLI allows the same URL to be used
```

Comment:
probably worth investigating separately

Reply:
Updated the comment, essentially if the same URL exists for multiple subgraphs, the previous entry in `subgraphsByURL` would be overridden, unsure if it causes any issue, but worth double checking.